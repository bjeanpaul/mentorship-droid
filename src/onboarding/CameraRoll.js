import React, { Component } from 'react';
import { View, Image, CameraRoll, StyleSheet, ScrollView} from 'react-native';
import { Link } from 'react-router-native';

import { Text, Heading, Button } from 'src/components';


const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#F5FCFF',
    },
    imageGrid: {
      flex: 1,
      flexDirection: 'row',
      flexWrap: 'wrap',
      justifyContent: 'center',
    },
    image: {
      width: 100,
      height: 100,
      margin: 10,
    },
});

class CameraRollPicker extends Component {

  constructor(props, context) {
    super(props);
    this.state = {
      photos: [],
    };

    this.nextPath = '/profile-picture';
    this.context = context;
  }

  componentDidMount() {
    CameraRoll.getPhotos({ first: 25 }).then((data) => {
      this.setState({ photos: data.edges.map((asset) => asset.node.image) });
    });
  }

  render() {
    return (
      <ScrollView style={styles.container}>
        <View style={styles.imageGrid}>
          {this.state.photos.map((photo, index) =>
            <Link key={`photo-${index}`} to={`${this.nextPath}?imageURI=${photo.uri}`}>
              <Image source={{ uri: photo.uri }} style={styles.image} />
            </Link>
          )}
        </View>
      </ScrollView>
    );
  }
}
CameraRollPicker.contextTypes = {
  router: React.PropTypes.object.isRequired,
};

export default CameraRollPicker;
