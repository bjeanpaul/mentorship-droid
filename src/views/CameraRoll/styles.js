import { StyleSheet } from 'react-native';


export default StyleSheet.create({
  imageGrid: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  imageContainer: {
    borderColor: '#000',
    margin: 1,
    width: 100,
    height: 100,
  },
  image: {
    flex: 1,
  },
});
