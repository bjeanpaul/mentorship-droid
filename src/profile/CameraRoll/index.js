import React, { PropTypes } from 'react';
import { View, Image, CameraRoll, ScrollView, TouchableNativeFeedback } from 'react-native';
import styles from './styles';


class CameraRollPicker extends React.Component {
  constructor(props) {
    super(props);

    const {
      initialPhotos = [],
    } = props;

    this.state = {
      photos: initialPhotos,
    };
  }

  componentDidMount() {
    return CameraRoll
      .getPhotos({ first: 25 })
      .then((data) => {
        this.setState({
          photos: this.state.photos
            .concat(data.edges.map((asset) => asset.node.image.uri)),
        });
      });
  }

  render() {
    return (
        <ScrollView style={{ flex: 1 }}>
          <View style={styles.imageGrid}>
            {this.state.photos.map((photoPath, index) =>
                <TouchableNativeFeedback
                  key={index}
                  photoId={index}
                  onPress={() => this.props.onPhotoPress(photoPath)}
                >
                  <View style={styles.imageContainer}>
                    <Image
                      source={{ uri: photoPath }}
                      style={styles.image}
                    />
                  </View>
                </TouchableNativeFeedback>
            )}
          </View>
        </ScrollView>
    );
  }
}

CameraRollPicker.propTypes = {
  onPhotoPress: PropTypes.func.isRequired,
  initialPhotos: PropTypes.array,
};

export default CameraRollPicker;
