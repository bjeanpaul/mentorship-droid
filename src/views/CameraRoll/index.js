import React, { PropTypes } from 'react';
import { View, Image, CameraRoll, ScrollView, TouchableNativeFeedback } from 'react-native';

import { BaseView, Header, Text } from 'src/components';
import styles from 'src/views/CameraRoll/styles';


class CameraRollPicker extends React.Component {
  constructor(props) {
    super(props);

    const {
      initialPhotos = [],
    } = props;

    this.state = {
      photos: initialPhotos,
    };
  }

  componentDidMount() {
    return CameraRoll
      .getPhotos({ first: 25 })
      .then((data) => {
        this.setState({
          photos: this.state.photos
            .concat(data.edges.map((asset) => asset.node.image.uri)),
        });
      });
  }

  render() {
    return (
      <BaseView>
        <Header>
          <Text style={Text.types.title}>Choose a photo</Text>
        </Header>
        <View style={styles.body}>
          <ScrollView >
            <View style={styles.imageGrid}>
              {this.state.photos.map((photoPath, index) =>
                  <TouchableNativeFeedback
                    key={index}
                    photoId={index}
                    onPress={() => this.props.onPhotoPress(photoPath)}
                  >
                    <View style={styles.imageContainer}>
                      <Image
                        source={{ uri: photoPath }}
                        style={styles.image}
                      />
                    </View>
                  </TouchableNativeFeedback>
              )}
            </View>
          </ScrollView>
        </View>
      </BaseView>
    );
  }
}

CameraRollPicker.propTypes = {
  onPhotoPress: PropTypes.func.isRequired,
  initialPhotos: PropTypes.array,
};

export default CameraRollPicker;
