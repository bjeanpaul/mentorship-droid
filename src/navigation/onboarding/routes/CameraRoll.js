import { connect } from 'react-redux';
import CameraRoll from 'src/profile/CameraRoll';
import * as actions from 'src/navigation/onboarding/actions';
import { changeProfileImage } from 'src/profile/actions';


const mapDispatchToProps = (dispatch) => {
  return {
    onPhotoPress: path => {
      dispatch(changeProfileImage(path));
      return dispatch(actions.popOnboardingRoute());
    },
  };
};

export default connect(null, mapDispatchToProps)(CameraRoll);
