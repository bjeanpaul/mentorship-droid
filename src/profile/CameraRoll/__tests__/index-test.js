import React from 'react';
import { shallow } from 'enzyme';
import CameraRoll from '../index';

describe('CameraRoll', () => {
  function noop() {}

  it('should map props correctly', () => {
    expect(
      <CameraRoll
        initialPhotos={[
          'path/to/photo1.png',
          'path/to/photo2.png',
          'path/to/photo3.png',
        ]}
        onPhotoPress={noop}
      />
    ).toMatchSnapshot();
  });

  it('should be able to tap and fire `onPhotoPress`', () => {
    const onPhotoPress = jest.fn();
    const el = shallow(
      <CameraRoll
        initialPhotos={[
          'path/to/photo1.png',
          'path/to/photo2.png',
          'path/to/photo3.png',
        ]}
        onPhotoPress={onPhotoPress}
      />
    );
    el.find('TouchableNativeFeedback').at(1).simulate('press');
    expect(onPhotoPress).toBeCalledWith('path/to/photo2.png');
  });
});
