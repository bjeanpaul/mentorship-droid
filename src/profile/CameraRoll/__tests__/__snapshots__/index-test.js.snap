exports[`CameraRoll should map props correctly 1`] = `
<ScrollView
  style={
    Object {
      "flex": 1
    }
  }>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap"
      }
    }>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300
          },
          undefined
        ]
      }>
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20
          }
        }>
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300
          },
          undefined
        ]
      }>
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20
          }
        }>
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300
          },
          undefined
        ]
      }>
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20
          }
        }>
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
  </View>
</ScrollView>
`;
